### PR TITLE
Support sample_weight in RMSE

### PR DESCRIPTION
**Pull request by jjaeyeon**
*Description of changes:*
- Supporting weighted RMSE for evaluation metric - if user specifies kwargs, it will use scikit learn's RMSE function, or else it will use the existing accelerated RMSE computation
- [tested] test-run with the example dataset

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/autogluon/core/metrics/__init__.py [hunk 1]
```diff
@@ -454,8 +454,11 @@ def local_pearsonr(y_true, y_pred):
                        greater_is_better=True)
 
 
-def rmse_func(y_true, y_pred):
-    return np.sqrt(((y_true - y_pred) ** 2).mean())
+def rmse_func(y_true, y_pred, **kwargs):
+    if kwargs:
+        return sklearn.metrics.mean_squared_error(y_true, y_pred, squared=False, **kwargs)
```
Comment (liangfu) [inline core/src/autogluon/core/metrics/__init__.py:459]: why squared is False ?

Comment (liangfu) [inline core/src/autogluon/core/metrics/__init__.py:459]: Why not use this directly and drop numpy version ?

**Review by liangfu — COMMENTED** [review]
Thanks for the contribution. Would be better if you could add a test case for this as well.

File: core/src/autogluon/core/metrics/__init__.py [hunk 2]
```diff
@@ -454,8 +454,11 @@ def local_pearsonr(y_true, y_pred):
                        greater_is_better=True)
 
 
-def rmse_func(y_true, y_pred):
-    return np.sqrt(((y_true - y_pred) ** 2).mean())
+def rmse_func(y_true, y_pred, **kwargs):
```
Comment (Innixma) [inline core/src/autogluon/core/metrics/__init__.py:457]: Please add a simple unit test that calls this function with and without sample weights, verifying that it returns the correct score. You can refer to the existing metric unit tests for log_loss and roc_auc: https://github.com/autogluon/autogluon/blob/master/core/tests/unittests/metrics/test_classification_metrics.py#L130-L222

File: core/src/autogluon/core/metrics/__init__.py (hunk 1 above)
Comment (jjaeyeon) [reply core/src/autogluon/core/metrics/__init__.py:459]: Discussed with Nick, following the implementation of roc_auc, we keep this to use accelerated (skipping validation checks) implementation if no additional argument is given

**Comment by liangfu** [comment]
> I see the test failure from timeseries due to tensors in different device - is this a known issue?

https://github.com/autogluon/autogluon/pull/3058 this has merged, and should fix the issue. Please rebase latest master branch.

**Review by Innixma — APPROVED** [review]
LGTM, thanks for the contribution!
